components/Map/Map.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import districtJson from '../../geoJSON/district.json';
+import subcountyJson from '../../geoJSON/subcounty.json';
+import * as turf from '@turf/turf';
+import * as distance from 'jaro-winkler';
+
+const style = {
+  width: '100%',
+  height: '600px'
+};
+
+interface MapProps {
+  district: string;
+  boundaryType: string;
+}
+
+interface State {
+  leaflet: {};
+  map: {};
+}
+
+class Map extends React.Component<MapProps, State> {
```

Comment:
Change whole thing to a function and use hooks ... then I'll review more.

Reply:
Do this for the other component as well.

Reply:
Updated `MapContainer` to use hooks